Style Guide stuff

ethanwright3 opened the pull request:
Brings up to style guide with things like the formatting and the references tab. Also cleans out a few packages. Proper DS package tables put in as well.

ethanwright3 committed:
Style Guide stuff

Brings up to style guide with things like the formatting and the references tab. Also cleans out a few packages. Proper DS package tables put in as well.

neilhatfield reviewed (dismissed):
Things to tackle:

1) Issue #3 
2) I feel that there are some high level issues that we need to discuss and resolve. Primarily,
    + What are the explicit goals and learning outcomes for this app?
    + What is the difference between a "simple" and an "advanced" data visualization?
    + If a goal is to "introduces how to create some commonly used plots [in R]", then we need to come up with the list of what commonly used __data visualizations__ are and design/build from there
    + What is the role/purpose of the Exercises tab?
3) Overview Page
    + Don't use `strong` on text inside of headings, e.g, `h2(strong("Instructions:"))`. Format these as `h2("Instructions")`
    + There shouldn't be an About heading. The first paragraph under the `h1` contains the about information.
    + You don't need the colons (:) in any headings (or titles) (goes for all pages, sweetAlerts, labels, etc.)
4) Visualization pages
    + Should we maintain "simple" vs "advanced", drop the word "Data" from the sidebar and make "Visualization" plural.
    + I think that I would prefer a page for Univariate visualizations, one for Bivariate visualizations, and one for Multivariate visualizations rather than the tab structure currently used. (Exercises would be their own page as well.)
    + We need to follow all Best Practices guidelines for plots (titles, axes, legends, readability, etc.)--This applies everywhere
    + This is not a mark against you, but I get the sense of "plot vomit" when I look at some the tabs. That is to say, there are suddenly plots on the page (some times multiple) with essentially nothing to ground them in a context, why we might want them, show how to make them, etc.
    + While I have comments on individual plots, I think we need to tackle higher level issues first.
    + I will note that the Select Colorscale for the Heatmaps should NOT be a slider. Make into a `selectInput`.
5) References should be alphabetized.
6) Code
    + Resolve long lines of code (i.e., lines with more than 80 characters)
    + Remove extraneous CSS code (e.g., line 43 in the ui)
    + Remove any comments that are not information (or make them informative)
    + Remove any commented out code that is no longer necessary
    + Format code for readability. For example, you have this
```
div(style = 'text-align: center',
                bsButton(inputId = 'go2', label = 'Explore',
                         icon = icon('bolt'), size = 'large', class='circle grow')),
```

which would be better as 
```
div(
  style = 'text-align: center;',
  bsButton(
    inputId = 'go2', 
    label = 'Explore',
    icon = icon('bolt'),
    size = 'large'
  )
),
```

Notice the addition of the semi-colon and removal of the class argument.

I think that these are enough things to tackle at the moment. Especially since the second one will (re-)shape the app.

ymy5178 reviewed (dismissed):
Recommended Priority: Medium

1. In my computer, I cannot run the code without changing rlocker to rLocker. 
2. There is no description file.
3. There is no description and link in the readme file.
4. In the two variable visualization, if the x and y can only choose continuous variables, then barplot and boxplot would be meaningless. 
<img width="631" alt="rlocker" src="https://user-images.githubusercontent.com/70001189/171091657-52ae1720-fb90-48ef-b880-7d582d52d26f.PNG">
<img width="633" alt="description" src="https://user-images.githubusercontent.com/70001189/171091789-38043c71-f9af-4bf1-b3d2-4e59355ecc73.PNG">

ymy5178 committed:
rlocker & reference & visulization

In this commit, I changed every rlocker to rLocker. And also, I corrected the sequence of reference. Also, I changed visulization to visualization.

JingFu1 reviewed (changes requested):
1. When running on my computer, the following appears :
Mapping API still under development and may change in future releases.
Warning: `line.width` does not currently support multiple values.
2. Missing references

Recommended Priority: Medium

pophillips reviewed (changes requested):
- File structure needs major revision. No docs folder. No app.R file (app is found in ui.R). Many additional files that need to be in www folder.
- README contains no required info, needs to be completely rewritten. All README items missing.
- app failed to run on my computer, though may be error on my end.
- Code needs to be relabeled with comments separating sections/pages (formatting of what looks like labels is wrong).

Recommended Priority: Medium

ymy5178 commented:
Hello Jing,

Can you help me to review the data visualization app again? I am so sorry for making branch confusion before.

Have a great day!
Best,
Yijun Yao


发件人: Jing Fu ***@***.***>
发送时间: 2022年7月7日 17:35
收件人: EducationShinyAppTeam/Data_Visualization ***@***.***>
抄送: Yao, Yijun ***@***.***>; Review requested ***@***.***>
主题: Re: [EducationShinyAppTeam/Data_Visualization] Style Guide stuff (#2)


@JingFu1 requested changes on this pull request.

  1.  When running on my computer, the following appears :
Mapping API still under development and may change in future releases.
Warning: line.width does not currently support multiple values.
  2.  Missing references

Recommended Priority: Medium

―
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FEducationShinyAppTeam%2FData_Visualization%2Fpull%2F2%23pullrequestreview-1032187076&data=05%7C01%7Cymy5178%40psu.edu%7Cdb53fe12964e473c36b608da60607f52%7C7cf48d453ddb4389a9c1c115526eb52e%7C0%7C0%7C637928264819423080%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=iYnMDOq%2F3yACdxLhiSFD4YNUqySniI9xQZ1oydoRB4E%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAQWCEJLS2UUMYCDT6USAVJTVS5EO5ANCNFSM4RSPBZIA&data=05%7C01%7Cymy5178%40psu.edu%7Cdb53fe12964e473c36b608da60607f52%7C7cf48d453ddb4389a9c1c115526eb52e%7C0%7C0%7C637928264819423080%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=P%2FQB0jPWHgk62r3bk%2FxXCWErNoSZIJMDOXW%2Bb8YHQ18%3D&reserved=0>.
You are receiving this because your review was requested.Message ID: ***@***.******@***.***>>

pophillips commented:
I will have a new review done by tomorrow afternoon!

pophillips reviewed (commented):
-Way too many misc. files
-Missing docs folder
-Rename ui.R to app.R
-code disorganized; needs to fit within margins

Recommend use other dev branch for all future edits/merging.
